storybook e2e tests fail to run with Storybook 6.2-alpha

<!-- Please do your best to fill out all of the sections below! -->

## Current Behavior
When I run `nx run testlib-e2e:e2e`, the test doesn't run. Instead  "Cannot find module '@storybook/core/dist/server/build-dev' " is printed to the console. See the "Failure Logs" section for the full error.

## Expected Behavior
Tests should run. Tests run properly when using the stable version of storybook (6.1).

I understand this is a prerelease of Storybook and there's no guarantee that nx will work with it, but I assume you want to be aware of these errors (that's what prereleases are for after all!)
I think the fix would be to change [this line](https://github.com/nrwl/nx/blob/master/packages/storybook/src/builders/storybook/storybook.impl.ts#L13) to import from `@storybook/core/server` instead of the longer path. A similar change would also be needed on [line 54 of the same file](https://github.com/nrwl/nx/blob/master/packages/storybook/src/builders/storybook/storybook.impl.ts#L13). (the hardest part will most likely be detecting the correct path to import from)

## Steps to Reproduce

[Use this repo](https://github.com/literalpie/nx-storybook-bug-cannot-find-build-dev-module) and skip to step 7.

1. Create an nx angular project
2. `nx generate library testlib`
3. `nx g c testcomp --project=testlib`
4. `yarn add --dev @nrwl/storybook`
5. `nx g @nrwl/angular:storybook-configuration testlib` - answer yes to all questions
6. `yarn add @storybook/angular@next @storybook/addon-knobs@next`
7. `nx run testlib-e2e:e2e` - This will cause the error

### Failure Logs
```
Cannot find module '@storybook/core/dist/server/build-dev'
Require stack:
- /Users/bkindle/Documents/Development/temp/testco/node_modules/@nrwl/storybook/src/builders/storybook/storybook.impl.js
- /Users/bkindle/Documents/Development/temp/testco/node_modules/@angular-devkit/architect/node/node-modules-architect-host.js
- /Users/bkindle/Documents/Development/temp/testco/node_modules/@angular-devkit/architect/node/index.js
- /Users/bkindle/Documents/Development/temp/testco/node_modules/@nrwl/tao/src/commands/ngcli-adapter.js
- /Users/bkindle/Documents/Development/temp/testco/node_modules/@nrwl/tao/src/commands/run.js
- /Users/bkindle/Documents/Development/temp/testco/node_modules/@nrwl/tao/index.js
- /Users/bkindle/Documents/Development/temp/testco/node_modules/@nrwl/cli/lib/run-cli.js
```

### Environment
  Node : 12.14.1
  OS   : darwin x64
  yarn : 1.22.10
  
  nx : Not Found
  @nrwl/angular : 11.2.10
  @nrwl/cli : 11.2.10
  @nrwl/cypress : 11.2.10
  @nrwl/devkit : 11.2.10
  @nrwl/eslint-plugin-nx : 11.2.10
  @nrwl/express : Not Found
  @nrwl/jest : 11.2.10
  @nrwl/linter : 11.2.10
  @nrwl/nest : Not Found
  @nrwl/next : Not Found
  @nrwl/node : Not Found
  @nrwl/react : Not Found
  @nrwl/schematics : Not Found
  @nrwl/tao : 11.2.10
  @nrwl/web : Not Found
  @nrwl/workspace : 11.2.10
  typescript : 4.0.5

--- a/package.json
+++ b/package.json
@@ -224,7 +224,7 @@
     "rxjs-for-await": "0.0.2",
     "sass": "1.26.3",
     "sass-loader": "8.0.2",
-    "semver": "6.3.0",
+    "semver": "7.3.4",
     "shelljs": "^0.8.3",
     "source-map": "0.7.3",
     "source-map-loader": "0.2.4",

--- a/packages/devkit/package.json
+++ b/packages/devkit/package.json
@@ -29,7 +29,7 @@
     "@nrwl/tao": "*",
     "ejs": "^3.1.5",
     "ignore": "^5.0.4",
-    "semver": "6.3.0",
+    "semver": "7.3.4",
     "strip-json-comments": "2.0.1",
     "tslib": "^2.0.0"
   }

--- a/packages/storybook/package.json
+++ b/packages/storybook/package.json
@@ -33,7 +33,7 @@
     "@nrwl/linter": "*",
     "@nrwl/workspace": "*",
     "core-js": "^3.6.5",
-    "semver": "6.3.0",
+    "semver": "7.3.4",
     "tree-kill": "1.2.2",
     "ts-loader": "5.4.5",
     "tsconfig-paths-webpack-plugin": "3.2.0",

--- a/packages/storybook/src/executors/build-storybook/build-storybook.impl.spec.ts
+++ b/packages/storybook/src/executors/build-storybook/build-storybook.impl.spec.ts
@@ -6,13 +6,19 @@ jest.mock('@storybook/core/standalone', () =>
 );
 import * as storybook from '@storybook/core/standalone';
 import storybookBuilder from './build-storybook.impl';
-
-import angularStorybookOptions from '@storybook/angular/dist/server/options';
+import * as fileUtils from '@nrwl/workspace/src/core/file-utils';
 
 describe('Build storybook', () => {
   let context: ExecutorContext;
 
   beforeEach(async () => {
+    jest.spyOn(fileUtils, 'readPackageJson').mockReturnValue({
+      devDependencies: {
+        '@storybook/addon-essentials': '^6.0.21',
+        '@storybook/angular': '^6.0.21',
+      },
+    });
+
     context = {
       root: '/root',
       cwd: '/root',

--- a/packages/storybook/src/executors/build-storybook/build-storybook.impl.ts
+++ b/packages/storybook/src/executors/build-storybook/build-storybook.impl.ts
@@ -4,7 +4,7 @@ import { constants, copyFileSync, mkdtempSync, statSync } from 'fs';
 
 import * as build from '@storybook/core/standalone';
 
-import { setStorybookAppProject } from '../utils';
+import { getStorybookFrameworkPath, setStorybookAppProject } from '../utils';
 import { ExecutorContext, logger } from '@nrwl/devkit';
 
 export interface StorybookConfig {
@@ -33,7 +33,8 @@ export default async function buildStorybookExecutor(
 ) {
   logger.info(`NX ui framework: ${options.uiFramework}`);
 
-  const frameworkPath = `${options.uiFramework}/dist/server/options`;
+  const frameworkPath = getStorybookFrameworkPath(options.uiFramework);
+
   const { default: frameworkOptions } = await import(frameworkPath);
   const option = storybookOptionMapper(options, frameworkOptions, context);
   logger.info(`NX Storybook builder starting ...`);

--- a/packages/storybook/src/executors/storybook/storybook.impl.spec.ts
+++ b/packages/storybook/src/executors/storybook/storybook.impl.spec.ts
@@ -4,6 +4,7 @@ jest.mock('@storybook/core/server', () => ({
   buildDevStandalone: jest.fn().mockImplementation(() => Promise.resolve()),
 }));
 import { buildDevStandalone } from '@storybook/core/server';
+import * as fileUtils from '@nrwl/workspace/src/core/file-utils';
 
 import { vol } from 'memfs';
 jest.mock('fs', () => require('memfs').fs);
@@ -14,6 +15,13 @@ describe('@nrwl/storybook:storybook', () => {
   let context: ExecutorContext;
   let options: StorybookExecutorOptions;
   beforeEach(() => {
+    jest.spyOn(fileUtils, 'readPackageJson').mockReturnValue({
+      devDependencies: {
+        '@storybook/addon-essentials': '^6.0.21',
+        '@storybook/angular': '^6.0.21',
+      },
+    });
+
     options = {
       uiFramework: '@storybook/angular',
       port: 4400,

--- a/packages/storybook/src/executors/storybook/storybook.impl.ts
+++ b/packages/storybook/src/executors/storybook/storybook.impl.ts
@@ -4,7 +4,7 @@ import { constants, copyFileSync, mkdtempSync, statSync } from 'fs';
 
 import { buildDevStandalone } from '@storybook/core/server';
 
-import { setStorybookAppProject } from '../utils';
+import { getStorybookFrameworkPath, setStorybookAppProject } from '../utils';
 import { ExecutorContext } from '@nrwl/devkit';
 
 export interface StorybookConfig {
@@ -37,7 +37,7 @@ export default async function* storybookExecutor(
   options: StorybookExecutorOptions,
   context: ExecutorContext
 ) {
-  const frameworkPath = `${options.uiFramework}/dist/server/options`;
+  let frameworkPath = getStorybookFrameworkPath(options.uiFramework);
 
   const frameworkOptions = (await import(frameworkPath)).default;
   const option = storybookOptionMapper(options, frameworkOptions, context);

--- a/packages/storybook/src/executors/utils.ts
+++ b/packages/storybook/src/executors/utils.ts
@@ -1,8 +1,39 @@
 import { ExecutorContext } from '@nrwl/devkit';
+import { readPackageJson } from '@nrwl/workspace/src/core/file-utils';
+import { gte } from 'semver';
 
 export interface NodePackage {
   name: string;
   version: string;
+}
+
+export function getStorybookFrameworkPath(uiFramework) {
+  const serverOptionsPaths = {
+    '@storybook/angular': '@storybook/angular/dist/ts3.9/server/options',
+    '@storybook/react': '@storybook/react/dist/cjs/server/options',
+  };
+
+  if (isStorybookV62onwards(uiFramework)) {
+    return serverOptionsPaths[uiFramework];
+  } else {
+    return `${uiFramework}/dist/server/options`;
+  }
+}
+
+function isStorybookV62onwards(uiFramework) {
+  const packageJsonContents = readPackageJson();
+  packageJsonContents.dependencies = packageJsonContents.dependencies || {};
+  packageJsonContents.devDependencies =
+    packageJsonContents.devDependencies || {};
+
+  const storybookPackageVersion =
+    packageJsonContents.dependencies[uiFramework] ||
+    packageJsonContents.devDependencies[uiFramework];
+
+  return gte(
+    (storybookPackageVersion || '').replace('~', '').replace('^', ''),
+    '6.2.0-rc.4'
+  );
 }
 
 // see: https://github.com/storybookjs/storybook/pull/12565

--- a/packages/tao/package.json
+++ b/packages/tao/package.json
@@ -34,7 +34,7 @@
     "minimist": "^1.2.5",
     "rxjs": "^6.5.4",
     "strip-json-comments": "2.0.1",
-    "semver": "6.3.0",
+    "semver": "7.3.4",
     "tmp": "0.0.33",
     "tslib": "^2.0.0",
     "yargs-parser": "20.0.0",

--- a/packages/web/package.json
+++ b/packages/web/package.json
@@ -94,7 +94,7 @@
     "rollup-plugin-typescript2": "^0.27.1",
     "sass": "^1.26.3",
     "sass-loader": "8.0.2",
-    "semver": "6.3.0",
+    "semver": "7.3.4",
     "source-map": "0.7.3",
     "source-map-loader": "0.2.4",
     "source-map-support": "0.5.16",

--- a/packages/workspace/package.json
+++ b/packages/workspace/package.json
@@ -65,7 +65,7 @@
     "npm-run-all": "^4.1.5",
     "opn": "^5.3.0",
     "rxjs": "^6.5.4",
-    "semver": "6.3.0",
+    "semver": "7.3.4",
     "strip-json-comments": "2.0.1",
     "tmp": "0.0.33",
     "yargs": "15.4.1",

